src/client/Microsoft.Identity.Client/Internal/Requests/ManagedIdentityAuthRequest.cs

```diff
@@ -181,24 +123,18 @@ private async Task<MsalAccessTokenCacheItem> GetCachedAccessTokenAsync()
             return null;
         }
 
-        private AuthenticationResult CreateAuthenticationResultFromCache(MsalAccessTokenCacheItem cachedAccessTokenItem)
-        {
-            AuthenticationResult authResult = new AuthenticationResult(
-                                                            cachedAccessTokenItem,
-                                                            null,
-                                                            AuthenticationRequestParameters.AuthenticationScheme,
-                                                            AuthenticationRequestParameters.RequestContext.CorrelationId,
-                                                            TokenSource.Cache,
-                                                            AuthenticationRequestParameters.RequestContext.ApiEvent,
-                                                            account: null,
-                                                            spaAuthCode: null,
-                                                            additionalResponseParameters: null);
-            return authResult;
-        }
+        protected abstract Task<AuthenticationResult> GetAccessTokenAsync(CancellationToken cancellationToken, ILoggerAdapter logger);
 
         protected override KeyValuePair<string, string>? GetCcsHeader(IDictionary<string, string> additionalBodyParameters)
         {
             return null;
         }
+
+        // Override method to return a sorted set of scopes based on the input set.
+        protected override SortedSet<string> GetOverriddenScopes(ISet<string> inputScopes)
```

Comment:
Why? There's only 1 scope possible? Why do we need to bring this whole "collection of scope" concept from user tokens to S2S tokens?